platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/merkle/VirtualRootNode.java

```diff
@@ -849,14 +861,42 @@ public boolean isMerged() {
      * If called, this copy of the map will eventually be flushed.
      */
     public void enableFlush() {
-        this.shouldBeFlushed = true;
+        this.shouldBeFlushed.set(true);
+    }
+
+    /**
+     * Sets flush threshold for this virtual root. When a copy of this virtual root is created,
+     * it inherits the threshold value.
+     *
+     * If this virtual root is explicitly marked to flush using {@link #enableFlush()}, changing
+     * flush threshold doesn't have any effect.
+     *
+     * @param value The flush threshold, in bytes
+     */
+    public void setFlushThreshold(long value) {
+        flushThreshold.getAndSet(value);
```

Comment:
We could just call flushThreshold.set(value), as we're not actually using the existing value here.

Reply:
Agree. Let me change it to plain `set()` here.